src/emulator/storage/apis/firebase.ts

```diff
@@ -193,16 +193,18 @@ export function createFirebaseEndpoints(emulator: StorageEmulator): Router {
   });
 
   const handleUpload = async (req: Request, res: Response) => {
-    if (!req.query.name) {
-      res.sendStatus(400);
-      return;
-    }
-
     const bucketId = req.params.bucketId;
-    const objectId = req.query.name.toString();
+    const objectId: string | null = req.params.objectId
```

Comment:
Could you separate this out into another pr or add some more description to this pr for what this fixes? I'm guessing you caught some discrepancies while running conformance tests but its not too clear

Reply:
Updated the PR description.

When creating a new object, prod supports the following url patterns:
* `v0/b/<bucket>/o/<new object>`
* `v0/b/<bucket>/o?name=<new object>`

This PR fixes the conformance test in `firebase.endpoints.test.ts` to ensure that the emulator supports both